lib/npm/verify-node-gyp.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+var uuid = require('node-uuid');
+var path = require('path');
+var fs = require('fs');
+var which = require('which').sync;
+
+function VerifyNodeGyp(context) {
+  if (!(this instanceof VerifyNodeGyp))
+    return new VerifyNodeGyp(context);
+  this.verifyNodeGyp = context.module.verifyNodeGyp;
+  var id = uuid.v4();
+  var scriptFilename = id + '.js';
+  this.script = path.resolve(context.path, scriptFilename);
+  this.marker = path.resolve(context.path, id + '.tmp');
+
+  var npmLocation = which('npm');
+  var npmNodeGyp = process.platform === 'win32' ?
+    path.resolve(path.dirname(npmLocation), 'node_modules', 'npm', 'bin', 'node-gyp-bin', 'node-gyp.cmd') :
+    path.resolve(path.dirname(npmLocation), '..', 'lib', 'node_modules', 'npm', 'bin', 'node-gyp-bin', 'node-gyp');
+  var script = '#!/usr/bin/env node\n' +
```

Comment:
I am not comfortable with the script being built line by line with inline text here. Is there another way this could be accomplished?